source/data-modeling/associations.txt

```diff
@@ -515,3 +515,410 @@ method to remove all embedded ``Album`` documents from the ``Band`` class:
+   :language: ruby
+   :start-after: # start-extensions
+   :end-before: # end-extensions
+   :emphasize-lines: 4-6
+
+Custom Association Names
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can use the ``class_name`` macro to specify a custom class name for an
+association. This is useful when you want to use a different name for an
+association than the name of the class. The following example uses the
+``class_name`` macro to specify that an embedded association called ``records``
+represents the ``Album`` class:
+
+.. literalinclude:: /includes/data-modeling/association-behaviors.rb
+   :language: ruby
+   :start-after: # start-custom-name
+   :end-before: # end-custom-name
+
+Custom Keys
+~~~~~~~~~~~
+
+By default, {+odm+} uses the ``_id`` field of the parent class when looking up
+associations. You can specify a different field to use by specifying the
+``primary_key`` and ``foreign_key`` macros. The following example specifies a new
+primary and foreign key for the ``albums`` association on a ``Band`` class:
+
+.. literalinclude:: /includes/data-modeling/association-behaviors.rb
+   :language: ruby
+   :start-after: # start-custom-keys
+   :end-before: # end-custom-keys
+
+If you are specifying a ``has_and_belongs_to_many`` association, you can also
+set the ``inverse_primary_key`` and ``inverse_foreign_key`` macros. The
+``inverse_primary_key`` macro specifies the field on the local model that the
+remote model uses to look up the documents.
+The ``inverse_foreign_key`` macro specifies the field on the remote model
+that stores the values found in ``inverse_primary_key``.
+
+The following example specifies a new primary and foreign key for the
+``Band`` and ``Members`` classes in a ``has_and_belongs_to_many`` association:
+
+.. literalinclude:: /includes/data-modeling/association-behaviors.rb
+   :language: ruby
+   :start-after: # start-custom-inverse-keys
+   :end-before: # end-custom-inverse-keys
+   :emphasize-lines: 9, 20
+
+Custom Scopes
+~~~~~~~~~~~~~
+
+You can set a specific scope on an association by using the ``scope`` parameter.
+The ``scope`` parameter determines which documents {+odm+} considers to be part
+of an association. A scoped association returns only documents that match the
+scope conditions when queried. You can set the ``scope`` to either a ``Proc`` with an arity
+of zero or a ``Symbol`` that references a named scope on the associated model.
+The following example sets custom scopes on associations in a ``Band`` class:
+
+.. literalinclude:: /includes/data-modeling/association-behaviors.rb
+   :language: ruby
+   :start-after: # start-custom-scope
+   :end-before: # end-custom-scope
+
+.. note::
+
+   You can add documents that do not match the scope conditions to an
+   association. {+odm+} saves the documents to the database and they will appear
+   in associated memory, however, the documents are not present when the
+   association is queried in the future.
```

Comment:
i: [comma splice](https://www.mongodb.com/docs/meta/style-guide/style/punctuation/#commas:~:text=Don%27t%20use%20only%20a%20comma%20to%20separate%20independent%20clauses.%20Doing%20so%20creates%20a%20comma%20splice.)

```suggestion
   association. {+odm+} saves these documents to the database and they will appear
   in associated memory, but the documents are not present when the
   association is queried in the future.
```

the last clause mixes tenses ('are not present' + 'in the future'), but i don't know enough to suggest a change. is there a way to do it in second-person (e.g., ' you won't see the documents if you query...')?

Reply:
Good call, reworded this with second-person